docs/fsharp/style-guide/formatting.md

```diff
@@ -1040,210 +852,408 @@ with
+This also applies to members, constructors, and parameters using tuples:
 
 ```fsharp
-let printListWithOffset a list1 =
-    List.iter
-        (fun elem ->
-             printfn $"A very long line to format the value: %d{a + elem}")
-        list1
+type TypeWithLongMethod() =
+    member _.LongMethodWithLotsOfParameters
+        (
+            aVeryLongParam: AVeryLongTypeThatYouNeedToUse,
+            aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse,
+            aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse
+        ) =
+        // ... the body of the method
+
+type TypeWithLongConstructor
+    (
+        aVeryLongCtorParam: AVeryLongTypeThatYouNeedToUse,
+        aSecondVeryLongCtorParam: AVeryLongTypeThatYouNeedToUse,
+        aThirdVeryLongCtorParam: AVeryLongTypeThatYouNeedToUse
+    ) =
+    // ... the body of the class follows
 ```
 
-If the lambda argument is the last argument in a function application, place all arguments until the arrow on the same line.
+If the parameters are curried, place the `=` character along with any return type on a new line:
 
 ```fsharp
-Target.create "Build" (fun ctx ->
-    // code
-    // here
-    ())
+type TypeWithLongCurriedMethods() =
+    member _.LongMethodWithLotsOfCurriedParamsAndReturnType
+        (aVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        (aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        (aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        : ReturnType =
+        // ... the body of the method
 
-let printListWithOffsetPiped a list1 =
-    list1
-    |> List.iter (fun elem ->
-        printfn $"A very long line to format the value: %d{a + elem}")
+    member _.LongMethodWithLotsOfCurriedParams
+        (aVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        (aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        (aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        =
+        // ... the body of the method
 ```
 
-Treat match lambda's in a similar fashion.
+This is a way to avoid too long lines (in case return type might have long name) and have less line-damage when adding parameters.
+
+### Formatting operator declarations
+
+Optionally use white space to surround an operator definition:
 
 ```fsharp
-functionName arg1 arg2 arg3 (function
-    | Choice1of2 x
-    | Choice2of2 y)
+// OK
+let ( !> ) x f = f x
+
+// OK
+let (!>) x f = f x
 ```
 
-Unless there are many leading or multiline arguments before the lambda. In that case, indent all arguments with one scope.
+For any custom operator that starts with `*` and that has more than one character, you need to add a white space to the beginning of the definition to avoid a compiler ambiguity. Because of this, we recommend that you simply surround the definitions of all operators with a single white-space character.
+
+## Formatting record declarations
```

Comment:
Since this is a declaration, should it be an H3 since it would fit under the broader "Formating declarations" section?